src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php

```diff
@@ -510,6 +515,11 @@ static function (): bool {
                     }
                 );
             }
+
+            $this->lastActivityTime = time();
+        } elseif (isset($this->connectionOptions['heartbeat']) && time() > $this->lastActivityTime + ($this->connectionOptions['heartbeat'] * 2)) {
```

Comment:
```suggestion
        } elseif ($this->connectionOptions['heartbeat'] > 0 && time() > $this->lastActivityTime + ($this->connectionOptions['heartbeat'] * 2)) {
```

Would that fix your concern @adegrenand?

Reply:
Considering that the heartbeat option as not a default value, this might get `Undefined array key "heartbeat"`

Reply:
@surfingmig yes we can use isset to avoid disconnection when a user don't pass a heartbeat. But if he pass 0 ( 0 means the server does not want a heartbeat) you will be always disconnected which implies a new connection each times to communicated with rabbitmq. 
```suggestion
        } elseif (isset($this->connectionOptions['heartbeat']) && $this->connectionOptions['heartbeat'] > 0 && time() > $this->lastActivityTime + ($this->connectionOptions['heartbeat'] * 2)) {
```